src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -0,0 +1,370 @@
+        // 
+        // Inputs: The AssemblyLoadContext and AssemblyName to be loaded
+        // Returns: The Loaded assembly object.
+        public event Func<AssemblyLoadContext, AssemblyName, Assembly> Resolving;
+
+        public event Action<AssemblyLoadContext> Unloading;
+
+        // Occurs when an Assembly is loaded
+        public static event AssemblyLoadEventHandler AssemblyLoad;
+
+        // Occurs when resolution of type fails
+        public static event ResolveEventHandler TypeResolve;
+
+        // Occurs when resolution of resource fails
+        public static event ResolveEventHandler ResourceResolve;
+
+        // Occurs when resolution of assembly fails
+        // This event is fired after resolve events of AssemblyLoadContext fails
+        public static event ResolveEventHandler AssemblyResolve;
+
+        public static AssemblyLoadContext Default => DefaultAssemblyLoadContext.s_loadContext;
+
+        public bool IsCollectible { get; }
+
+        // Helper to return AssemblyName corresponding to the path of an IL assembly
+        public static AssemblyName GetAssemblyName(string assemblyPath)
+        {
+            if (assemblyPath == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyPath));
+            }
+
+            return AssemblyName.GetAssemblyName(assemblyPath);
+        }
+
+        // Custom AssemblyLoadContext implementations can override this
+        // method to perform custom processing and use one of the protected
+        // helpers above to load the assembly.
+        protected abstract Assembly Load(AssemblyName assemblyName);
+
+        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
+        public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
+        {
+            if (assemblyName == null)
+                throw new ArgumentNullException(nameof(assemblyName));
+
+            // Attempt to load the assembly, using the same ordering as static load, in the current load context.
+            StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
+            return Assembly.Load(assemblyName, ref stackMark, _nativeAssemblyLoadContext);
+        }
+
+        // These methods load assemblies into the current AssemblyLoadContext 
+        // They may be used in the implementation of an AssemblyLoadContext derivation
+        public Assembly LoadFromAssemblyPath(string assemblyPath)
+        {
+            if (assemblyPath == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyPath));
+            }
+
+            if (PathInternal.IsPartiallyQualified(assemblyPath))
+            {
+                throw new ArgumentException(SR.Argument_AbsolutePathRequired, nameof(assemblyPath));
+            }
+
+            lock (_unloadLock)
+            {
+                VerifyIsAlive();
```

Comment:
This is breaking a corefx test: previously if both a partial path was specified and the context was unloaded, we'd get an InvalidOperationException, because this check came first, but now we get an ArgumentException because it comes second.  @janvorli, should I fix the ordering here or should I fix up the test to match this expected ordering?

Reply:
I would fix the test, the new ordering seems more logical to me.

Reply:
Ok, thanks.